### ISSUE TITLE
[OPD] [megatron/fsdp] use forward_kl_topk method occurs oom

### ISSUE THREAD

**Issue by chenjiaoAngel** [verl-project/verl#6810]
### System Info

- 基于example/opd scripts 运行OPD 测试，在32xh20(96G) 下；使用最新main 分支code测试（06.18）运行forward_kl_topk + topk=32 在vocab_parallel_log_softmax 计算student 和 teacher logits 出现OOM；
- 在forward_kl_topk 模式下，use_fused_kernel 需要设置为false，要不model output 输出没有找不到distillation_losses

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

student=Qwen3-8b teacher=qwen3-32b
response_len=20K
batch_size=128
topk=32
trian: tp4pp1cp2
infer: tp2
student=2x8=16 卡H20
teacher=2x8=16 卡H20
遇到OOM：TP 加大，response_len 减少，仍然没有用
<img width="2576" height="464" alt="Image" src="https://github.com/user-attachments/assets/c05898f3-1e67-425a-818f-08ff1f4de6f5" />

### Expected behavior

- 可以正常运行，不会出现oom
- 支持use_fused_kernel=True 下，正常运行

**Comment by yzhbradoodrrpurp** [comment]
我也遇到了类似的情况
